File: random_dog_flutter/lib/features/randomdog/domain/usecases/random_dog_image.dart
import 'package:random_dog_flutter/common/data/responses/local_response.dart';
import 'package:random_dog_flutter/features/randomdog/domain/entities/dog_image.dart';
import 'package:random_dog_flutter/features/randomdog/domain/repositories/random_dog_repository.dart';

class RandomDogImage {
  final RandomDogRepository repository;
  RandomDogImage(this.repository);

  Future<LocalResponse<DogImage>> randomImage(int number) async {
    return await repository.randomImage(number);
  }
}

File: KMM/shared/src/commonMain/kotlin/com/kkunan/kmm/randomdog/domain/usecases/RandomDogImage.kt
package com.kkunan.kmm.randomdog.domain.usecases

import com.kkunan.kmm.randomdog.domain.entities.DogImage
import com.kkunan.kmm.randomdog.domain.repositories.DogImageRepository

class RandomDogImage(private val repository: DogImageRepository) {
    suspend operator fun invoke(): DogImage?{
        val response = repository.randomImage(1)
        return response?.let { images ->
            if (images.isEmpty()){
                null
            } else {
                images[0]
            }
        }
    }
}